firefox-ios/Client/Coordinators/Router/RouteBuilder.swift

```diff
@@ -155,7 +156,12 @@ final class RouteBuilder: FeatureFlaggable {
 
     func makeRoute(userActivity: NSUserActivity) -> Route? {
         // If the user activity is a Siri shortcut to open the app, show a new search tab.
-        if userActivity.activityType == SiriShortcuts.activityType.openURL.rawValue {
+        // By using shouldOpenNewTab we avoid duplicated user activities, from Siri, for new tab.
+        if userActivity.activityType == SiriShortcuts.activityType.openURL.rawValue && shouldOpenNewTab {
+            shouldOpenNewTab = false
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1) { [weak self] in
+                self?.shouldOpenNewTab = true
+            }
```

Comment:
Can we have unit tests for this? 🙏 `DispatchQueue.main.asyncAfter` can be tested by using `DispatchQueueInterface`